libflux/src/libstd/lib.rs

```diff
@@ -19,6 +24,50 @@ pub fn fresher() -> Fresher {
     flatbuffers::get_root::<fb::Fresher>(buf).into()
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn flux_semantic_analyze(
+    src_ptr: *const c_char,
+    flux_buf: *mut flux_buffer_t,
+) -> *mut flux_error_t {
+    let buf = CStr::from_ptr(src_ptr).to_bytes(); // Unsafe
+    let s = String::from_utf8(buf.to_vec()).unwrap();
+    match analyze(s.as_str()) {
+        Ok(vec) => {
+            let flux_buf = &mut *flux_buf;
+            flux_buf.data = vec.as_ptr();
+            flux_buf.len = vec.len();
+            std::mem::forget(vec);
+            std::ptr::null_mut()
+        }
+        Err(err) => {
+            let errh = flux::ErrorHandle { err: Box::new(err) };
+            return Box::into_raw(Box::new(errh)) as *mut flux_error_t;
+        }
+    }
+}
+
+fn analyze(src: &str) -> Result<Vec<u8>, flux::Error> {
+    let mut f = fresher();
+
+    let ast_file = flux::parser::parse_string("", src);
```

Comment:
Do we want to let the caller determine the file name of the returned ast.File? Then if the caller wants an empty string they just pass one in?